section_faiss_vector_index.go

```diff
@@ -216,6 +234,16 @@ func (v *vectorIndexOpaque) flushVectorSection(vecToDocID map[int64]uint64,
 	return fieldStart, nil
 }
 
+// Calculates the nprobe count, given nlist(number of centroids) based on
+// the metric the index is optimized for.
+func calculateNprobe(nlist int, indexOptimizedFor string) int32 {
+	nprobe := int32(math.Sqrt(float64(nlist)))
+	if indexOptimizedFor == index.IndexOptimizedForLatency {
+		nprobe /= 2
```

Comment:
Must handle the case where nprobe is less than 2 - in which case here we'd be setting it to zero.

Reply:
The thresholds in determineIndexType() were chosen to ensure that there are never less than 10 centroids(1000/100) so the minimum values of nprobe would be around 3.
Still added a check to be on the safe side but just something to keep in mind.